MoreLinq/ExceptBy.cs

```diff
@@ -24,6 +24,54 @@ namespace MoreLinq
     static partial class MoreEnumerable
     {
         /// <summary>
+        /// Returns a distinct set of elements from the first collection,
+        /// whose keys are not in the second collection.
+        /// </summary>
+        /// <typeparam name="T">The type of source, keys, and result elements.</typeparam>
+        /// <param name="first">The source collection.</param>
+        /// <param name="second">The key collection.</param>
+        /// <param name="keySelector">The key selector.</param>
+        /// <returns>
+        /// Distinct set of elements from first collection,
+        /// whose keys are not in second collection.
+        /// </returns>
+        public static IEnumerable<T> ExceptBy<T>(this IEnumerable<T> first,
+            IEnumerable<T> second,
+            Func<T, T> keySelector) {
```

Comment:
Funny you should say that, actually.  

I don't think this overload is particularly useful, its really just a special case of when ```TKey``` and ```TSource``` are the same. In this case ```keySelector``` is a projection from one T to another T, not the identity function.  

Originally I did not have this overload (or the corresponding one on IntersectBy). After adding the overload with this signature:

```IEnumerable<TSource> ExceptBy<TSource, TKey>(this IEnumerable<TSource> first, IEnumerable<TKey> second, Func<TSource, TKey> keySelector)```

several of the unit tests for ExceptBy that were using String as both TSource and TKey gave me ambiguous type errors.  I added the ```ExceptBy<T>``` overload (and explicit type parameters in several methods on ExceptByTest) to get rid of those errors.

I believe my original overload is very useful, but I would like to get rid of ```ExceptBy<T>``` if possible.  I've been using this new ```ExceptBy<TSource, TKey>``` overload in practice now for a while, and I've never had ambiguous type issues.  I will see if there is a way to alter the tests to remove the ambiguity without adding any additional overloads.